### PR TITLE
[CALCITE-3226] Expand view should keep the alias expanded view 

### PR DESCRIPTION
**Pull request by jinxing64**
In current implementation of `RelBuilder::scan ` (https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/tools/RelBuilder.java#L1048), the alias can be derived and recorded into `Frame` only when the `RelNode` is a `TableScan` (https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/tools/RelBuilder.java#L2754).
But when `RelBuilder::scan` from an expanded view, the node is not a `TableScan` and the alias is not kept.  Below test failed - we cannot reference a field by alias "MYVIEW".
```
  @Test public void testExpandViewShouldKeepAlias() throws SQLException {
    try (Connection connection = DriverManager.getConnection("jdbc:calcite:")) {
      final Frameworks.ConfigBuilder configBuilder =
          expandingConfig(connection);
      final RelOptTable.ViewExpander viewExpander =
          (RelOptTable.ViewExpander) Frameworks.getPlanner(configBuilder.build());
      final RelFactories.TableScanFactory tableScanFactory =
          RelFactories.expandingScanFactory(viewExpander,
              RelFactories.DEFAULT_TABLE_SCAN_FACTORY);
      configBuilder.context(Contexts.of(tableScanFactory));
      final RelBuilder builder = RelBuilder.create(configBuilder.build());
      RelNode node =
          builder.scan("MYVIEW")
              .project(
                  builder.field(1, "MYVIEW", "EMPNO"),                    // Exception thrown from here.
                  builder.field(1, "MYVIEW", "ENAME"))
              .build();
      String expected =
          "LogicalProject(EMPNO=[$0], ENAME=[$1])\n"
              + "  LogicalFilter(condition=[=(1, 1)])\n"
                  + "    LogicalTableScan(table=[[scott, EMP]])\n";
      assertThat(node, hasTree(expected));
    }
  }
```

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/tools/RelBuilder.java [hunk 1]
```diff
@@ -1054,6 +1054,11 @@ public RelBuilder scan(Iterable<String> tableNames) {
     final RelNode scan = scanFactory.createScan(cluster, relOptTable);
     push(scan);
     rename(relOptTable.getRowType().getFieldNames());
+
+    // When the node is not a TableScan but from expansion, we add the alias.
```
Comment (rubenada) [inline core/src/main/java/org/apache/calcite/tools/RelBuilder.java]: I would rephrase a bit the comment line, for example something like:
"..., we need to explicitly add the alias."

**Comment by rubenada** [comment]
@jinxing64 I'm +1 on the proposed solution.
However, I'm not 100% sure if there might be another place (maybe "cleaner" or "more correct") where this special case could/should be handled. So I prefer to have more opinions on the topic before merging.

**Comment by jinxing64** [comment]
@rubenada 
Thanks a lot for review ~
Yes, I will follow your suggestion :)

**Comment by rubenada** [comment]
@jinxing64 , since @julianhyde gave his ok on the jira ticket, I'm fine with moving on and merging the patch. Could you please squash your commits into a single one? Thanks

**Comment by jinxing64** [comment]
Thanks a lot for approve @rubenada 
I squashed the commits

**Comment by jinxing64** [comment]
THX for merging ! @rubenada
